src/main/java/org/elasticsearch/cluster/block/ClusterBlockLevel.java

```diff
@@ -44,13 +53,18 @@ public int id() {
         return this.id;
     }
 
-    public static ClusterBlockLevel fromId(int id) {
+    public static EnumSet<ClusterBlockLevel> fromId(int id) {
         if (id == 0) {
-            return READ;
+            return EnumSet.of(READ);
         } else if (id == 1) {
-            return WRITE;
+            return EnumSet.of(WRITE);
         } else if (id == 2) {
-            return METADATA;
+            // After 1.5.0 METADATA has been splitted into two distincts cluster block levels
+            return EnumSet.of(METADATA_READ, METADATA_WRITE);
+        } else if (id == 3) {
+            return EnumSet.of(METADATA_READ);
+        } else if (id == 4) {
+            return EnumSet.of(METADATA_WRITE);
```

Comment:
this `id` gets sent over the wire in `ClusterBlock#readFrom` and `ClusterBlock#writeTo`. your change makes it backwards compatible only for reads, cause a 1.6 node that gets 1.2 detects and converts it. But what happens if a newer node sends 3 or 4 to an older node? We need to add some logic based on version of nodes.

Also, I'd try and make this method package private, not sure why it's public it shouldn't IMO.